601 refactor filter functionality for table views

DuciDuc03 opened the pull request:
## DESCRIPTION

_description of the scope of this PR_

_loom link if applicable_

### TO-DO

- [x] implement
- [x] pull `main` & resolve merge conflicts

### CLOSES

closes _list of issues closed in this PR_

qhantom reviewed (commented):
Works great in itself! I just noticed that when I type into a filter input, it's a bit laggy. Can you take another look at this? Presumably the whole page is being re-rendered.

DuciDuc03 commented:
@qhantom 
I removed the loader from the table component during filtering. Additionally, I implemented a debounced filter for users in the user view. Previously, without the debounce, the table would briefly show no entries while filtering, which caused the entire table to jump unexpectedly.

qhantom reviewed (approved):
Nice one!
